src/controllers/user.controller.ts

```diff
@@ -154,25 +173,58 @@ export const getProfile = async (req: Request, res: Response): Promise<void> =>
   }
 };
 
-
 export const updateProfile = async (req: Request, res: Response) => {
   try {
-    const updatedFields= req.body;
-    const {id} =req.params;
+    const updatedFields = req.body;
+    const { id } = req.params;
 
-    const newProfile = await modifyProfile({updatedFields, id});
-    return res.status(201).json({ message: "profile updated!", data: newProfile });
+    const newProfile = await modifyProfile({ updatedFields, id });
+    return res
+      .status(201)
+      .json({ message: 'profile updated!', data: newProfile });
   } catch (error) {
-    let message
-    if (error instanceof Error) message = error.message
-    else message = String(error)
+    let message;
+    if (error instanceof Error) message = error.message;
+    else message = String(error);
 
     if (message.includes('no fields to update')) {
       return res.status(400).json({ message: 'no fields to update' });
-    } else if  (message.includes('profile not found')) {
+    } else if (message.includes('profile not found')) {
       return res.status(400).json({ message: 'profile not found' });
     } else {
-      return res.status(500).json({ message: 'Internal server error', error:message });
+      return res
+        .status(500)
+        .json({ message: 'Internal server error', error: message });
     }
   }
 };
+export const disableUserAccount = async (req: CustomRequest, res: Response) => {
+  try {
+    const { user_id } = req.params;
+    const user = await UserService.disableUser(user_id);
+
+    if (typeof user === 'string') {
+      return res.status(200).json({ message: user });
+    }
```

Comment:
If this is successful... how will the code block below be executed when you have returned here?